### PR TITLE
"Working with invalid data" added (Anti-pattern documentation)

### PR DESCRIPTION
**Pull request by lucasvegi**
I only filled in the TODO for "Working with invalid data" to validate the template. Once I receive confirmation, I will gradually continue filling in the others.

### CODE REVIEW

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 1]
```diff
@@ -13,7 +13,65 @@ TODO
...
+  # with invalid data errors are easy to locate and fix
+  iex(2)> MyApp.foo("José")
+  ** (FunctionClauseError) no function clause matching in MyApp.foo/1
+
+    The following arguments were given to MyApp.foo/1:
+
+        # 1
+        "José"
+
+    my_app.ex:6: MyApp.foo/1
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:72]:
```suggestion
  defmodule MyLibrary do
    def foo(data) when is_integer(data) do
      ...
      MyLibrary.Internal.sum(1, data)
      ...
    end
  end

  #...Use examples...

  #with valid data is ok
  iex(1)> MyLibrary.foo(2)
  3

  # with invalid data errors are easy to locate and fix
  iex(2)> MyLibrary.foo("José")
  ** (FunctionClauseError) no function clause matching in MyLibrary.foo/1

    The following arguments were given to MyLibrary.foo/1:

        # 1
        "José"

    my_library.ex:6: MyLibrary.foo/1
```

**Comment by josevalim** [comment]
@lucasvegi I pushed some small suggestions. It is not only about working with third party libraries but also your own code, so I have made it more specific (and added a comment about third-party).

**Comment by josevalim** [comment]
Also, can you please use `**` for bolding? And only a single backtick for code. That's the style we use in our codebase. :) Thank you!

**Comment by josevalim** [comment]
@lucasvegi this PR was closed because I merged the other PR. Please send it again pointing to main (sorry, i didn't know github would do that!).

**Comment by lucasvegi** [comment]
> @lucasvegi this PR was closed because I merged the other PR. Please send it again pointing to main (sorry, i didn't know github would do that!).

No problem. I have already opened a new one in the main branch (https://github.com/elixir-lang/elixir/pull/12753) and took the opportunity to make the suggested style formatting corrections. Can all future PRs be made directly on the main branch as well?

**Comment by josevalim** [comment]
Yes, all directly against main!

**Pull request closed without merging**